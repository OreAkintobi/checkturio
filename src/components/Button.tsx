import { TouchableOpacity, StyleSheet, Text } from 'react-native';
import { theme } from '../utils';

interface ButtonProps {
  title: string;
  onPress: () => void;
}

export const Button: React.FC<ButtonProps> = ({ title, onPress }) => {
  return (
    <TouchableOpacity style={styles.container} onPress={onPress}>
      <Text style={styles.text}>{title}</Text>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  container: {
    borderWidth: 1,
    borderColor: theme.colors.primary,
    borderRadius: theme.sizes.xl,
    paddingVertical: theme.sizes.m,
    paddingHorizontal: theme.sizes.xl,
    marginVertical: theme.sizes.m,
    marginRight: theme.sizes.xl,
    backgroundColor: theme.colors.background,
  },
  text: {
    color: theme.colors.primary,
    fontSize: theme.sizes.xl,
    textAlign: 'center',
    fontWeight: 'bold',
    marginVertical: theme.sizes.m,
  },
});
