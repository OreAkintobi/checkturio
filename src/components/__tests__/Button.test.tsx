import React from 'react';
import renderer from 'react-test-renderer';
import { TouchableOpacity } from 'react-native';
import { Button } from '../Button';

describe('Button', () => {
  it('renders correctly with title', () => {
    const mockOnPress = jest.fn();
    const tree = renderer
      .create(<Button title="Submit" onPress={mockOnPress} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('simulates press event', () => {
    const mockOnPress = jest.fn();
    const component = renderer.create(
      <Button title="Submit" onPress={mockOnPress} />
    );
    const buttonInstance = component.root.findByType(TouchableOpacity);

    // Simulate press event
    buttonInstance.props.onPress();

    // Assert onPress handler is called
    expect(mockOnPress).toHaveBeenCalled();
  });
});
