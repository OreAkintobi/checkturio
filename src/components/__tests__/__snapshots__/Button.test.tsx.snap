// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly with title 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "#ffffff",
      "borderColor": "#007bff",
      "borderRadius": 16,
      "borderWidth": 1,
      "marginRight": 16,
      "marginVertical": 8,
      "opacity": 1,
      "paddingHorizontal": 16,
      "paddingVertical": 8,
    }
  }
>
  <Text
    style={
      {
        "color": "#007bff",
        "fontSize": 16,
        "fontWeight": "bold",
        "marginVertical": 8,
        "textAlign": "center",
      }
    }
  >
    Submit
  </Text>
</View>
`;
